web-registry/src/lib/queries/react-query/ecocredit/getAddDataToBatchesQuery/getAddDataToBatchesQuery.types.ts

```diff
@@ -0,0 +1,18 @@
+import { QueryClient, QueryObserverOptions } from '@tanstack/react-query';
+
+import { BatchInfoWithSupply } from 'types/ledger/ecocredit';
+import { AddDataToBatchesParams } from 'lib/ecocredit/api';
+
+import { ReactQueryBuilderResponse } from '../../types/react-query.types';
+
+export type ReactQueryAddDataToBatchesResponse = QueryObserverOptions<
+  BatchInfoWithSupply[] | void
+>;
+
+export type ReactQueryAddDataToBatchesParams = Omit<
+  AddDataToBatchesParams,
+  'batches'
+> & {
+  batches?: AddDataToBatchesParams['batches'];
+  reactQueryClient?: QueryClient;
+} & ReactQueryBuilderResponse<ReactQueryAddDataToBatchesResponse>;
```

Comment:
`ReactQueryBuilderResponse` is a small ts utils to simplify query creation.

Reply:
i'm having some trouble understanding how this part works, but i'm curious to understand it better. maybe it's something you can cover tomorrow in the engineering check-in or some other time, definitely not a blocker